shotover-proxy/src/transforms/cassandra/sink_cluster/topology.rs

```diff
@@ -183,6 +227,174 @@ async fn fetch_current_nodes(
+                        replication
+                    )))
+                }
+            };
+
+            let replication_factor = match replication_strategy {
+                ReplicationStrategy::SimpleStrategy { replication_factor } => replication_factor,
+                ReplicationStrategy::NetworkTopologyStrategy {
+                    datacenter_replication_factor,
+                } => *datacenter_replication_factor.get(data_center).unwrap_or(&0),
+                _ => 0,
+            };
+
+            KeyspaceMetadata { replication_factor }
+        } else {
+            return Err(anyhow!(
+                "system_schema.keyspaces.replication is not a varchar"
+            ));
+        };
+
+        let name = if let Some(MessageValue::Varchar(name)) = row.pop() {
+            name
+        } else {
+            return Err(anyhow!("system_schema_keyspaces.name"));
+        };
+
+        Ok((name, metadata))
+    }
+
+    fn build_replication_strategy(
+        mut properties: Map<String, JsonValue>,
+    ) -> Result<ReplicationStrategy> {
+        match properties.remove("class") {
+            Some(JsonValue::String(class)) => Ok(match class.as_str() {
+                "org.apache.cassandra.locator.SimpleStrategy" | "SimpleStrategy" => {
+                    ReplicationStrategy::SimpleStrategy {
+                        replication_factor: extract_replication_factor(
+                            properties.get("replication_factor"),
+                        )?,
+                    }
+                }
+                "org.apache.cassandra.locator.NetworkTopologyStrategy"
+                | "NetworkTopologyStrategy" => ReplicationStrategy::NetworkTopologyStrategy {
+                    datacenter_replication_factor: extract_datacenter_replication_factor(
+                        properties,
+                    )?,
+                },
+                _ => ReplicationStrategy::Other,
+            }),
+            _ => Err(anyhow!("Missing replication strategy class")),
+        }
+    }
+
+    fn extract_datacenter_replication_factor(
+        properties: Map<String, JsonValue>,
+    ) -> Result<HashMap<String, usize>> {
+        properties
+            .into_iter()
+            .map(|(key, replication_factor)| {
+                extract_replication_factor(Some(&replication_factor))
+                    .map(move |replication_factor| (key, replication_factor))
+            })
+            .try_collect()
+    }
+
+    fn extract_replication_factor(value: Option<&JsonValue>) -> Result<usize> {
+        match value {
+            Some(JsonValue::String(replication_factor)) => {
+                let result = if let Some(slash) = replication_factor.find('/') {
+                    usize::from_str(&replication_factor[..slash])
+                } else {
+                    usize::from_str(replication_factor)
+                };
+
+                result.map_err(|error| {
```

Comment:
This should use context 
![image](https://user-images.githubusercontent.com/5120858/199660734-34a635ec-f5d3-4fc5-bf1e-1c12c73795ae.png)